File: flutter_app/lib/models/food.dart
// lib/models/food.dart
class Food {
  final String id;
  final String externalId;
  final String name;
  final double calories;
  final double protein;
  final double carbs;
  final double fat;

  Food({
    required this.id,
    required this.externalId,
    required this.name,
    required this.calories,
    required this.protein,
    required this.carbs,
    required this.fat,
  });

  factory Food.fromJson(Map<String, dynamic> json) {
    return Food(
      id: json['_id'] as String,
      externalId: json['externalId'] as String,
      name: json['name'] as String,
      calories: (json['calories'] as num).toDouble(),
      protein: (json['protein'] as num).toDouble(),
      carbs: (json['carbs'] as num).toDouble(),
      fat: (json['fat'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'externalId': externalId,
      'name': name,
      'calories': calories,
      'protein': protein,
      'carbs': carbs,
      'fat': fat,
    };
  }
}

File: backend/src/models/Food.ts
// src/models/Food.ts
import { Schema, model, Document } from 'mongoose';

export interface IFood extends Document {
  externalId: string; // ID en la API externa (p.ej. USDA)
  name: string;
  calories: number;
  protein: number;
  carbs: number;
  fat: number;
  createdAt: Date;
  updatedAt: Date;
}

const FoodSchema = new Schema<IFood>(
  {
    externalId: { type: String, required: true, unique: true },
    name: { type: String, required: true, index: true },
    calories: { type: Number, required: true },
    protein: { type: Number, required: true },
    carbs: { type: Number, required: true },
    fat: { type: Number, required: true },
  },
  { timestamps: true },
);

// Índices para búsquedas por nombre
FoodSchema.index({ name: 'text' });

export default model<IFood>('Food', FoodSchema);
